src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -131,91 +134,78 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
         // If aLocationName is not null, create a CharSpan to reference the text.
-        // Make sure there is always a terminating character, mainly so we can  get c_str for logging.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
         memset(mLocationNameBuffer, 0, sizeof(mLocationNameBuffer+1));
 
         // this assumes locationInfo structure was created above, if appropriate
         if (!locationInfo.locationInfo.IsNull())
         {
-            if (aLocationName.size() == 0)
+            if (aLocationName.empty())
             {
                 locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, 0);
             }
             else if (aLocationName.size() > sizeof(mLocationNameBuffer))
             {
-                // save truncated name that fits into available size
+                // Save the truncated name that fits into available size.
                 memcpy(mLocationNameBuffer, aLocationName.data(), sizeof(mLocationNameBuffer));
                 locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, sizeof(mLocationNameBuffer));
             }
             else
             {
-                // save full name
+                // Save full name.
                 memcpy(mLocationNameBuffer, aLocationName.data(), aLocationName.size());
                 locationInfo.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, aLocationName.size());
             }
         }
     }
 
     /**
-     * @brief Compare the location's name with given text
-     * @param[in] aLocationName name to compare
-     * @return true if location structure's name field matches aLocationName
+     * @brief Compare the location's name with the given text.
+     * @param[in] aLocationName The name to compare.
+     * @return true if the location structure's name field matches aLocationName.
      * 
-     * @note  if locations structure's name field is null, returns false
+     * @note if locations structure's name field is null, returns false.
      */
     bool DoesNameMatch(const CharSpan & aLocationName) const
     {
-        if ((!locationInfo.locationInfo.IsNull())   &&
-            (locationInfo.locationInfo.Value().locationName.size() == aLocationName.size()))
-        {
-            if (locationInfo.locationInfo.Value().locationName.size() == 0)
-            {
-                return true;  // both "empty" string
-            }
-            else
-            {
-                return (0 == memcmp(locationInfo.locationInfo.Value().locationName.data(), aLocationName.data(), 
-                                        locationInfo.locationInfo.Value().locationName.size()));
-            }
-        }
-        else
+        if (!locationInfo.locationInfo.IsNull())
         {
-            return false;  // name is Null or text sizes do not match
+            return locationInfo.locationInfo->locationName.data_equal(aLocationName);
```

Comment:
Looks good. I did this early on, was uncomfortable about  Span, more comfortable just doing it myself.
Looks like Span handles everything.
`    bool data_equal(const Span<const T> & other) const
    {
        return (size() == other.size()) && (empty() || (memcmp(data(), other.data(), size() * sizeof(T)) == 0));
    }`

Reply:
Yeah. I did look at the `data_equal` impl and it looked sound.